src/transactions/reducer.test.ts

```diff
@@ -0,0 +1,156 @@
+    value: '',
+  },
+  transactionHash: '0xfe6882fbcabc01debb1c4776c14f545d5ffd6877c460ac5fb6bcafab60089f3d',
+  block: '26861364',
+  type: TokenTransactionTypeV2.CrossChainSwapTransaction,
+  networkId: NetworkId['celo-mainnet'],
+  timestamp: 1721978706000,
+  __typename: 'CrossChainTokenExchange',
+}
+
+describe('transactions reducer', () => {
+  it('should handle UPDATE_TRANSACTIONS action for a pending cross chain swap', () => {
+    const state: State = {
+      ...initialState,
+      standbyTransactions: [standbyCrossChainSwap],
+    }
+    const updatedState = reducer(state, {
+      type: Actions.UPDATE_TRANSACTIONS,
+      networkId: NetworkId['celo-mainnet'],
+      transactions: [incomingCrossChainSwap],
+    })
+
+    expect(updatedState).toEqual({
+      ...initialState,
+      transactionsByNetworkId: {
+        [NetworkId['celo-mainnet']]: [], // the incoming pending transaction is not added to the store
+      },
+      standbyTransactions: [
+        // resulting standby transaction is the incoming pending transaction,
+        // plus the standby properties that we want to retain
+        {
+          ...incomingCrossChainSwap,
+          feeCurrencyId: 'celo-mainnet:native',
+          inAmount: {
+            value: '0.000030353037245945',
+            tokenId: 'op-mainnet:native',
+          },
+          context: {
+            id: '8440fcf7-11f4-45e9-874e-7a1198f9810e',
+            tag: 'swap/saga',
+            description: 'Swap/Execute',
+          },
+        },
+      ],
+    })
+  })
+
+  it('should handle UPDATE_TRANSACTIONS action for a pending cross chain swap that is not already in the store', () => {
+    const state: State = {
+      ...initialState,
+    }
+    const updatedState = reducer(state, {
+      type: Actions.UPDATE_TRANSACTIONS,
+      networkId: NetworkId['celo-mainnet'],
+      transactions: [incomingCrossChainSwap],
+    })
+
+    expect(updatedState).toEqual({
+      ...initialState,
+      transactionsByNetworkId: {
+        [NetworkId['celo-mainnet']]: [incomingCrossChainSwap],
+      },
+      standbyTransactions: [], // the incoming pending transaction is not added to the standby transactions
+    })
+  })
+
+  it('should handle UPDATE_TRANSACTIONS action for a completed cross chain swap', () => {
+    const state: State = {
+      ...initialState,
+      standbyTransactions: [standbyCrossChainSwap],
+    }
+    const updatedState = reducer(state, {
+      type: Actions.UPDATE_TRANSACTIONS,
+      networkId: NetworkId['celo-mainnet'],
+      transactions: [{ ...incomingCrossChainSwap, status: TransactionStatus.Complete }],
+    })
+
+    expect(updatedState).toEqual({
+      ...initialState,
+      transactionsByNetworkId: {
+        [NetworkId['celo-mainnet']]: [
+          { ...incomingCrossChainSwap, status: TransactionStatus.Complete },
+        ],
+      },
+      standbyTransactions: [standbyCrossChainSwap], // the standby transaction is not altered
```

Comment:
Shouldn't the standby TX be removed then?
Or will we do that only once we are able to get the inAmount from the explorer TX?

Reply:
the standby tx will be removed indeed, but by the cleanupStandbyTransactions saga (not by this reducer). i'm not sure that it makes sense to mention the sagas in the reducers test. or perhaps the confusion points to a bigger question of why we are modifying state via a saga _and_ reducer? i think that's a legitimate question actually, perhaps i can axe the saga entirely...